File: 90-SubsetsIi/90-SubsetsIi.cpp
// Last updated: 6/11/2026, 11:36:59 AM
class Solution {
public:
    set<vector<int>> ans;
    void rec(int ind, vector<int> temp, vector<int>& nums){
        if(ind == nums.size()){
            ans.insert(temp);
            return;
        }
        temp.push_back(nums[ind]);
        rec(ind+1, temp, nums);
        temp.pop_back();
        rec(ind+1, temp, nums);
    }

    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<int> temp;
        sort(nums.begin(), nums.end());
        ans.clear();
        rec(0, temp, nums);
        vector<vector<int>> ans2(ans.begin(), ans.end());
        return ans2;
    }
};

File: 90-SubsetsIi/90-SubsetsIi.py
# Last updated: 6/11/2026, 11:37:03 AM
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        ans = set()   # tuple(list)
        ans2 = []
        def rec(path, ind):
            if(ind==len(nums)):
                if(tuple(path) in ans):
                    return
                ans.add(tuple(path))
                ans2.append(path)
                print(path)
                return

            rec(path+[nums[ind]], ind+1)
            rec(path, ind+1)

        rec([], 0)
        return ans2